tests/test_streaming_download_manager.py

```diff
@@ -538,3 +542,33 @@ def test_streaming_gg_drive_zipped():
     assert xbasename(all_files[0]) == TEST_GG_DRIVE_FILENAME
     with xopen(all_files[0]) as f:
         assert f.read() == TEST_GG_DRIVE_CONTENT
+
+
+@pytest.mark.parametrize(
+    "urlpath, yield_type, file_names, file_contents",
+    [
+        (TEST_GG_DRIVE_ZIPPED_URL, tuple, TEST_GG_DRIVE_FILENAME, TEST_GG_DRIVE_CONTENT),
+        (TEST_GG_DRIVE2_ZIPPED_URL, tuple, TEST_GG_DRIVE2_FILENAME, TEST_GG_DRIVE2_CONTENT),
+    ],
+)
+def test_streaming_download_manager_iter_archive(urlpath, yield_type, file_names, file_contents):
+    file_names = file_names if isinstance(file_names, list) else [file_names]
+    file_contents = file_contents if isinstance(file_contents, list) else [file_contents]
+
+    archive_iterator = StreamingDownloadManager().iter_archive(urlpath)
+    files = [elm for elm in archive_iterator]
+    for file in files:
+        assert isinstance(file, yield_type)
+        assert file[0] in file_names
+        assert file[1].read().decode("utf-8") in file_contents
+
+
+@pytest.mark.parametrize(
+    "urlpath",
+    [(TEST_GG_DRIVE_GZIPPED_URL)],
+)
+@pytest.mark.xfail()
```

Comment:
It's ok if `iter_archive` on a gzip file raises `zipfile.BadZipFile` as specified in your test, so I would remove `xfail`

```suggestion
```